### ISSUE TITLE
[feature request] CPE search support

### ISSUE THREAD

**Issue by dogasantos** [opencve/opencve#92]
Hey guys,
First of all, thanks for the initiative. 

I see this evolving to become a de facto monitoring system and as such, a few features would be necessary.
There fore, i'd like to request such features in order to make it fit on most scenarios: 

**CPE search support via frontend and api**

To date, is not possible to use a cpe code straight from nmap or any other tool, like the following: cpe:/a:microsoft:internet_information_server

at least, it doesn't work at all on opencve.io and also self-hosted instances. 

This will allow us to integrate with a wide range of tools and make opencve the center piece. 

Thanks

**Comment by ncrocfer** [comment]
Yes it's definitively something we need to add. Nevertheless it's not so simple because each CVE has several associated configurations, including boolean expressions (and / or).

I think (but it's just a first step,I haven't thought much about it yet) we can not just send a CPE and get back a list of CVE, sometimes a CPE is vulnerable only if other conditions are met.

But in any case the need is here ;)

**Comment by dogasantos** [comment]
Yeap, it's a tricky part!
But thanks for the response! Really appreciated

**Comment by dogasantos** [comment]
Hi, sorry.
I've closed this by mistake.

**Comment by dagobertdebug** [comment]
Hi, that feature would have many benefits! Thanks for any efforts on this!

**Comment by dogasantos** [comment]
Ill give it a study and see if i can do something about
